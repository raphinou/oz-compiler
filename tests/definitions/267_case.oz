% Test !Vars in patterns

local
   R=rec(a b c)
   A=a
   B=b
in
   case R
   of rec(a !B c) andthen B==b then
      {Show matchb}  %<--
   [] rec(!A !B c) then
      {Show matchab}
   [] rec(a b c) then
      {Show matchrecord}
   else
      {Show nomatch}
   end

   case R
   of rec(a !A c) andthen B==b then
      {Show matchb}
   [] rec(!A !B c) then
      {Show matchab}   %<--
   [] rec(a b c) then
      {Show matchrecord}
   else
      {Show nomatch}
   end

   case R
   of rec(a !A c) andthen B==b then
      {Show matchb}
   [] rec(!A !A c) then
      {Show matchab}
   [] rec(a b c) then
      {Show matchrecord}    %<--
   else
      {Show nomatch}
   end

   case R
   of rec(a !A c) andthen B==b then
      {Show matchb}
   [] rec(!B !A c) then
      {Show matchab}
   [] rec(a b c) then
      {Show matchrecord}    %<--
   else
      {Show nomatch}
   end

   case R
   of rec(a !A c) andthen B==b then
      {Show matchb}
   [] rec(!B !A c) then
      {Show matchab}
   [] rec(a b d) then
      {Show matchrecord}
   else
      {Show nomatch}    %<--
   end
end
