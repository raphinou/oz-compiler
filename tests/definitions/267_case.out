matchb
matchab
matchrecord
matchrecord
nomatch
